### DISCUSSION TITLE
Autoryzacja WebApp ze Storage

### DISCUSSION THREAD

**Discussion by adriantheking** [AzureCommunityPL/group#127] [Pomoc] [answered]
Cześć,

Potrzebuję porady od bardziej doświadczonych kolegów.
Na chwilę obecną mam następujące rozwiązanie:
Wchodząc na webapp generowany jest SAS token i pobierane są pliki ze storage account, jednak przy dużej ilości plików samo generowanie tokenów i zwrócenie tego na front zajmuje trochę za dużo czasu. 

Potrzebuję jakiejś porady czy istnieje możliwość by pliki ze storage account były dostępne bez tokenów tylko gdy odpytuje po to WebApp, zaznaczę, że nie chcę ustawiać Anonimowego dostępu do storage.

**Comment by wi5nia** [comment]
SAS generalnie generuje zazwyczaj po to jak chcesz dac zabezpieczony i okreslony w czasie dostep bezposrednio do storage [klientowi](https://docs.microsoft.com/en-us/azure/storage/common/storage-sas-overview).

Wiec jesli Twoja webapka nie dostarcza uzytkownikom takich linkow tylko ten storage jest na potrzeby Twojej apki to poprostu korzystac z access key i trzymaj go np w KV albo w Configuration w WebApp jako connecion string i sie do niego odwoluj w kodziej.

Chyba, ze masz jeszcze jakis inny scenariusz to opisz dokladnie prosze.

**Comment by maciejgos** [comment]
Jak chcesz ograniczyć dostęp do storage/blob tylko dla twojej aplikacji to możesz użyć AAD https://docs.microsoft.com/en-us/azure/storage/common/storage-auth-aad?toc=/azure/storage/blobs/toc.json. Tak jak napisał @wi5nia SAS token służy do czegoś innego.

**Reply by mgrabarz** [reply]
Ciekawy scenariusz :)

**Comment by mgrabarz** [comment, marked as answer]
Odpowiem nieco przewrotnie...

Możesz twój AppService wpiać do VNetu poprzez VNet Integration. Oznacza to, że AppService będzie widoczny z publicznego Internetu, ale będzie miał możliwość "sięgania" do zasobów w VNet. Drugim krokiem jest utworzenie PrivateEndpointa dla Storage i wpięcie go do tego samego VNetu (ale drugiego subnetu). Storage zostanie wtedy kompletnie odcięty od świata i jego jedynym "wejściem" będzie wspomniany VNet. W efekcie AppService będzie mógł sięgać do Storage i TYLKO on.

Powyższy scenariusz spełnia wymaganie "tylko gdy odpytuje po to WebApp", jednak zakładam, że Ty chcesz "tylko javascript, z poziomu przeglądarki, wygenerowany przez WebApp". W takim scenariuszu pozostaje generowanie tokenów na backendzie i z tymi tokenami js wali prostu w Storage, albo generowanie czegokolwiek na backendzie, javascript wali do ciebie, weryfikujesz "cokolwiek" i po PrimaryKey sięgasz do Storage. To oznacza jednak, ze przepychasz te pliki przez swój serwer http.

**Reply by adriantheking** [reply]
Dokładnie o coś takiego mi chodziło. 
Scenariusz jest taki, że aplikacja trzecia wykonuje swoje prace i uploaduje obrazy na storage w bardzo dużych ilościach, następnie obrazy są wyświetlane na froncie, odbywało to się teraz właśnie przez generowanie tokenów i zwracanie ich na front - jednak jak wspominałem wcześniej przy dużej ilości obrazów generowanie zajmowało zbyt dużo czasu.

**Reply by wi5nia** [reply]
Idealny przyklad tego jak dokladny opis scenariusza, wymagan jest wazny w uzyskaniu odpowiedniej odpowiedzi 😄
